src/cleanup.ts

```diff
@@ -3,48 +3,79 @@ import { Utils } from './utils';
 
 async function cleanup() {
     if (!addCachedJfToPath()) {
-        core.error('Could not find JFrog CLI path in the step state. Skipping cleanup.');
+        core.warning('Could not find JFrog CLI executable. Skipping cleanup.');
         return;
     }
+    await buildInfoPostTasks();
+
+    // Cleanup JFrog CLI servers configuration
+    try {
+        core.startGroup('Cleanup JFrog CLI servers configuration');
+        await Utils.removeJFrogServers();
+    } catch (error) {
+        core.setFailed((<any>error).message);
+    } finally {
+        core.endGroup();
+    }
+}
+
+async function buildInfoPostTasks() {
+    const disableAutoBuildPublish: boolean = core.getBooleanInput(Utils.AUTO_BUILD_PUBLISH_DISABLE);
+    const disableJobSummary: boolean = core.getBooleanInput(Utils.JOB_SUMMARY_DISABLE);
+    if (disableAutoBuildPublish && disableJobSummary) {
+        core.info(`Both ${Utils.AUTO_BUILD_PUBLISH_DISABLE} and ${Utils.JOB_SUMMARY_DISABLE} are set to true. Skipping Build Info post tasks.`);
+        return;
+    }
+
+    // Check connection to Artifactory before proceeding with build info post tasks
+    try {
+        core.startGroup('Checking connection to JFrog Artifactory');
+        const pingResult: string = await Utils.runCliAndGetOutput(['rt', 'ping']);
+        if (pingResult !== 'OK') {
+            core.warning('Could not connect to Artifactory. Skipping Build Info post tasks.');
+        }
+    } catch (error) {
+        core.warning(`An error occurred while trying to connect to Artifactory: ${error}. Skipping Build Info post tasks.`);
+        return;
+    } finally {
+        core.endGroup();
+    }
+
     // Auto-publish build info if needed
     try {
-        if (!core.getBooleanInput(Utils.AUTO_BUILD_PUBLISH_DISABLE)) {
+        if (!disableAutoBuildPublish) {
             await collectAndPublishBuildInfoIfNeeded();
+        } else {
+            core.info('Auto build info publish is disabled. Skipping auto build info collection and publishing');
         }
     } catch (error) {
-        core.warning('failed while attempting to publish build info: ' + error);
+        core.warning('failed while attempting to collect and publish build info: ' + error);
     }
+
     // Generate job summary
     try {
-        if (!core.getBooleanInput(Utils.JOB_SUMMARY_DISABLE)) {
+        if (!disableJobSummary) {
             core.startGroup('Generating Job Summary');
             await Utils.runCli(['generate-summary-markdown']);
             await Utils.setMarkdownAsJobSummary();
             core.endGroup();
```

Comment:
This should be placed in a `finally` block. Let's utilize a utility that encapsulates the logic to execute code within a log group.

For instance:
```typescript
/**
 * Run a function inside a log group
 * @param name    - The name of the group
 * @param fn      - The logic to run
 * @param failure - The logic to run in case of failure
 */
function runInGroup(name: string, fn: () => Promise<void>, failure: (error: any) => Promise<void>) {
    try {
        core.startGroup(name);
        fn();
    } catch (error) {
        failure(error);
    } finally {
        core.endGroup();
    }
}
```